src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1856,7 +1856,7 @@ internal DeclarationTable Declarations
         /// </summary>
         public override ImmutableArray<Diagnostic> GetParseDiagnostics(CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
I think adding parameter to control filtering on GetParseDiagnostics are more natural.